Python/dynload_shlib.c

```diff
@@ -106,7 +106,7 @@ _PyImport_FindSharedFuncptr(const char *prefix,
         const char *error = dlerror();
         if (error == NULL)
             error = "unknown dlopen() error";
-        error_ob = PyUnicode_FromString(error);
+        error_ob = PyUnicode_DecodeLocale(error, "surrogateescape");
```

Comment:
Can we use `PyUnicode_DecodeFSDefault` instead?
Error message may contain file path.